diff --git a/node_modules/@architect/deploy/src/sam/index.js b/node_modules/@architect/deploy/src/sam/index.js
index ff8fb1d..3cfbc46 100644
--- a/node_modules/@architect/deploy/src/sam/index.js
+++ b/node_modules/@architect/deploy/src/sam/index.js
@@ -55,7 +55,7 @@ module.exports = function samDeploy (params, callback) {
   let compat, finalCloudFormation
 
   if (name) {
-    stackname += toLogicalID(name)
+    stackname += toLogicalID(name.toString())
   }
 
   if (eject) {
diff --git a/node_modules/@architect/deploy/src/static/publish/s3/put-files/index.js b/node_modules/@architect/deploy/src/static/publish/s3/put-files/index.js
index 6c1429f..e887046 100644
--- a/node_modules/@architect/deploy/src/static/publish/s3/put-files/index.js
+++ b/node_modules/@architect/deploy/src/static/publish/s3/put-files/index.js
@@ -1,7 +1,7 @@
 let { readFileSync } = require('fs')
 let crypto = require('crypto')
 let chalk = require('chalk')
-let series = require('run-series')
+let parallel = require('run-parallel-limit')
 let formatKey = require('./format-key')
 let putParams = require('./put-params')
 
@@ -16,7 +16,8 @@ module.exports = function putFiles (params, callback) {
     region,
     s3,
     staticManifest,
-    verbose
+    verbose,
+    maxParallel = 10
   } = params
 
   let uploaded = 0
@@ -96,7 +97,7 @@ module.exports = function putFiles (params, callback) {
 
   // Upload all the objects!
   // This used to be a parallel op, but large batches could rate limit out, so let's not
-  series(tasks, (err) => {
+  parallel(tasks, maxParallel, (err) => {
     if (err) callback(err)
     else callback(null, uploaded, notModified)
   })
