diff --git a/node_modules/@architect/architect/node_modules/@architect/deploy/src/static/publish/filter-files.js b/node_modules/@architect/architect/node_modules/@architect/deploy/src/static/publish/filter-files.js
index 791e0e3..592b379 100644
--- a/node_modules/@architect/architect/node_modules/@architect/deploy/src/static/publish/filter-files.js
+++ b/node_modules/@architect/architect/node_modules/@architect/deploy/src/static/publish/filter-files.js
@@ -21,8 +21,8 @@ module.exports = function filterFiles (params, callback) {
   // Sort for user readability
   filtered = sort(filtered)
 
-  // Sort again to ensure index.html files publish last
-  let index = /(index\.html?)$/
+  // Sort again to ensure index.html and service-worker.js files publish last
+  let index = /(index\.html?|service-worker\.js)$/
   filtered = filtered.sort((fileA, fileB) => {
     if (fileA.match(index) &&
         fileB.match(index)) return fileB.length - fileA.length
