diff --git a/node_modules/@architect/asap/src/lib/binary-extensions.js b/node_modules/@architect/asap/src/lib/binary-extensions.js
index 4415e42..83c6e0f 100644
--- a/node_modules/@architect/asap/src/lib/binary-extensions.js
+++ b/node_modules/@architect/asap/src/lib/binary-extensions.js
@@ -1,2 +1,2 @@
 // This file is auto-generated via npm run vendor
-module.exports = [ '3g2', '3gp', '3gpp', '7z', 'aif', 'aifc', 'aiff', 'au', 'avi', 'azw', 'bin', 'bmp', 'boz', 'bpk', 'buffer', 'bz', 'bz2', 'deb', 'deploy', 'dist', 'distz', 'dll', 'dmg', 'dms', 'doc', 'docx', 'dot', 'dump', 'ear', 'elc', 'epub', 'exe', 'gif', 'h264', 'heic', 'heics', 'heif', 'heifs', 'ico', 'img', 'iso', 'jar', 'jpe', 'jpeg', 'jpg', 'lrf', 'm1v', 'm2a', 'm2v', 'm3a', 'mar', 'mov', 'mp2', 'mp2a', 'mp3', 'mp4', 'mp4v', 'mpe', 'mpeg', 'mpg', 'mpg4', 'mpga', 'mpkg', 'msi', 'msm', 'msp', 'oga', 'ogg', 'ogv', 'opus', 'otf', 'pdf', 'pkg', 'png', 'pot', 'pps', 'ppt', 'pptx', 'qt', 'rar', 'rtf', 'snd', 'so', 'spx', 'tar', 'tif', 'tiff', 'ts', 'war', 'wav', 'webm', 'webp', 'woff', 'woff2', 'xla', 'xlc', 'xlm', 'xls', 'xlsx', 'xlt', 'xlw', 'zip' ]
+module.exports = [ '3g2', '3gp', '3gpp', '7z', 'aif', 'aifc', 'aiff', 'au', 'avi', 'azw', 'bin', 'bmp', 'boz', 'bpk', 'buffer', 'bz', 'bz2', 'deb', 'deploy', 'dist', 'distz', 'dll', 'dmg', 'dms', 'doc', 'docx', 'dot', 'dump', 'ear', 'elc', 'epub', 'exe', 'gif', 'h264', 'heic', 'heics', 'heif', 'heifs', 'ico', 'img', 'iso', 'jar', 'jpe', 'jpeg', 'jpg', 'lrf', 'm1v', 'm2a', 'm2v', 'm3a', 'mar', 'mov', 'mp2', 'mp2a', 'mp3', 'mp4', 'mp4v', 'mpe', 'mpeg', 'mpg', 'mpg4', 'mpga', 'mpkg', 'msi', 'msm', 'msp', 'oga', 'ogg', 'ogv', 'opus', 'otf', 'pdf', 'pkg', 'png', 'pot', 'pps', 'ppt', 'pptx', 'qt', 'rar', 'rtf', 'snd', 'so', 'spx', 'tar', 'tif', 'tiff', 'ts', 'war', 'wav', 'webm', 'webp', 'woff', 'woff2', 'xla', 'xlc', 'xlm', 'xls', 'xlsx', 'xlt', 'xlw', 'zip', 'wasm' ]
diff --git a/node_modules/@architect/asap/src/lib/binary-types.js b/node_modules/@architect/asap/src/lib/binary-types.js
index 0b66eb6..d576239 100644
--- a/node_modules/@architect/asap/src/lib/binary-types.js
+++ b/node_modules/@architect/asap/src/lib/binary-types.js
@@ -13,6 +13,7 @@ module.exports = [
   'application/vnd.openxmlformats-officedocument.presentationml.presentation',
   'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
   'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+  'application/wasm',
   // Fonts
   'font/otf',
   'font/woff',
