diff --git a/node_modules/@architect/functions/src/discovery/index.js b/node_modules/@architect/functions/src/discovery/index.js
index 8ceac52..f4d1f04 100644
--- a/node_modules/@architect/functions/src/discovery/index.js
+++ b/node_modules/@architect/functions/src/discovery/index.js
@@ -22,7 +22,7 @@ module.exports = function lookup (callback) {
   if (local) {
     let port = 2222
     if (ARC_SANDBOX) {
-      let { ports } = JSON.parse(ARC_SANDBOX)
+      let { ports = { _arc: 2222} } = JSON.parse(ARC_SANDBOX || '{}')
       if (!ports._arc) {
         return callback(ReferenceError('Sandbox internal port not found'))
       }
diff --git a/node_modules/@architect/functions/src/index.js b/node_modules/@architect/functions/src/index.js
index e0378ac..9e14bd8 100644
--- a/node_modules/@architect/functions/src/index.js
+++ b/node_modules/@architect/functions/src/index.js
@@ -13,9 +13,6 @@ if (!ARC_ENV) {
 if (!validEnvs.includes(ARC_ENV)) {
   throw ReferenceError(`ARC_ENV env var is required for use with @architect/functions`)
 }
-if (ARC_SANDBOX && !sandboxVersionAtLeast('5.0.0')) {
-  throw ReferenceError('Incompatible version: please upgrade to Sandbox >=5.x or Architect >=10.x')
-}
 
 let http = require('./http')
 let _static = require('./static')
diff --git a/node_modules/@architect/functions/src/ws/index.js b/node_modules/@architect/functions/src/ws/index.js
index caa0dbe..3384644 100644
--- a/node_modules/@architect/functions/src/ws/index.js
+++ b/node_modules/@architect/functions/src/ws/index.js
@@ -13,8 +13,8 @@ function instantiateAPI () {
   } = process.env
   let local = ARC_ENV === 'testing' || ARC_LOCAL
   if (local) {
-    let { ports } = JSON.parse(ARC_SANDBOX)
-    let port = ports._arc
+    let { ports } = JSON.parse(ARC_SANDBOX || '{}')
+    let port = ports?._arc || 2222
     if (!port) throw ReferenceError('Architect internal port not found')
     _api = new ApiGatewayManagementApi({
       apiVersion: '2018-11-29',
@@ -30,6 +30,10 @@ function instantiateAPI () {
   }
 }
 
+function isBuffer(buf) {
+  return Buffer.isBuffer(buf)
+}
+
 /**
  * arc.ws.send
  *
@@ -45,7 +49,7 @@ function send ({ id, payload }, callback) {
   instantiateAPI()
   let params = {
     ConnectionId: id,
-    Data: JSON.stringify(payload)
+    Data: isBuffer(payload) ? payload : JSON.stringify(payload)
   }
 
   if (callback) {
