diff --git a/node_modules/@nasa-gcn/architect-plugin-search/index.js b/node_modules/@nasa-gcn/architect-plugin-search/index.js
index f91368a..7eaeaae 100644
--- a/node_modules/@nasa-gcn/architect-plugin-search/index.js
+++ b/node_modules/@nasa-gcn/architect-plugin-search/index.js
@@ -510,6 +510,9 @@ async function executeSearchRequests(cwd) {
 var deploy = {
   // @ts-expect-error: The Architect plugins API has no type definitions.
   start({ cloudformation, inventory, arc, stage }) {
+    if (process.env.DEPOY_NAME !== 'alpha' && process.env.DEPOY_NAME !== 'dev') {
+      return cloudformation;
+    }
     let resources;
     const config = getConfig(arc);
     if (config.availabilityZoneCount) {
@@ -524,8 +527,13 @@ var deploy = {
   },
   // @ts-expect-error: The Architect plugins API has no type definitions.
   services({ stage, arc }) {
-    if (stage !== "production") {
-      return { node: "http://localhost:9200" };
+    if (process.env.DEPOY_NAME !== 'alpha' && process.env.DEPOY_NAME !== 'dev') {
+      var port = parseInt(process.env.DECI_SEARCH_PORT ?? '9200');
+      return { node: "http://localhost:" + port };
+    }
+    if (stage !== "production" && stage !== "staging") {
+      var port = parseInt(process.env.DECI_SEARCH_PORT ?? '9200');
+      return { node: "http://localhost:" + port };
     } else if (arc.search) {
       return services2;
     } else {
@@ -555,13 +563,13 @@ var sandbox = {
       }
     }
   }) {
-    const engine = getEngine(getConfig(arc).sandboxEngine);
-    local = await launch({ engine });
+    // const engine = getEngine(getConfig(arc).sandboxEngine);
+    // local = await launch({ engine });
     await executeSearchRequests(cwd);
-    await populate(cwd, { node: local.url });
+    // await populate(cwd, { node: local.url });
   },
   async end() {
-    await local.stop();
+    // empty
   }
 };
 export {
