diff --git a/node_modules/@nasa-gcn/architect-plugin-search/index.js b/node_modules/@nasa-gcn/architect-plugin-search/index.js
index f91368a..858311b 100644
--- a/node_modules/@nasa-gcn/architect-plugin-search/index.js
+++ b/node_modules/@nasa-gcn/architect-plugin-search/index.js
@@ -524,8 +524,9 @@ var deploy = {
   },
   // @ts-expect-error: The Architect plugins API has no type definitions.
   services({ stage, arc }) {
-    if (stage !== "production") {
-      return { node: "http://localhost:9200" };
+    if (stage !== "production" && stage !== "staging") {
+      var port = parseInt(process.env.DECI_SEARCH_PORT ?? '9200');
+      return { node: "http://localhost:" + port };
     } else if (arc.search) {
       return services2;
     } else {
@@ -555,13 +556,13 @@ var sandbox = {
       }
     }
   }) {
-    const engine = getEngine(getConfig(arc).sandboxEngine);
-    local = await launch({ engine });
+    // const engine = getEngine(getConfig(arc).sandboxEngine);
+    // local = await launch({ engine });
     await executeSearchRequests(cwd);
-    await populate(cwd, { node: local.url });
+    // await populate(cwd, { node: local.url });
   },
   async end() {
-    await local.stop();
+    // empty
   }
 };
 export {
