diff --git a/node_modules/@typescript-eslint/eslint-plugin/dist/rules/consistent-type-imports.js b/node_modules/@typescript-eslint/eslint-plugin/dist/rules/consistent-type-imports.js
index 67bbcbd..84315f1 100644
--- a/node_modules/@typescript-eslint/eslint-plugin/dist/rules/consistent-type-imports.js
+++ b/node_modules/@typescript-eslint/eslint-plugin/dist/rules/consistent-type-imports.js
@@ -263,7 +263,7 @@ exports.default = (0, util_1.createRule)({
                              * ```
                              * https://github.com/typescript-eslint/typescript-eslint/issues/7527
                              */
-                            if (report.node.attributes.length === 0) {
+                            if (!report.node.attributes?.length) {
                                 context.report({
                                     node: report.node,
                                     messageId: 'typeOverValue',
