diff --git a/node_modules/@udecode/plate-autoformat/dist/index.es.js b/node_modules/@udecode/plate-autoformat/dist/index.es.js
index ef14dcf..6281155 100644
--- a/node_modules/@udecode/plate-autoformat/dist/index.es.js
+++ b/node_modules/@udecode/plate-autoformat/dist/index.es.js
@@ -148,9 +148,13 @@ const autoformatBlock = (editor, {
       if (isBelowSameBlockType) continue;
     }
 
-    Transforms.delete(editor, {
-      at: matchRange
-    });
+    // if the trigger is only 1 char there is nothing to delete, so we'd delete unrelated text
+    if (match.length > 1) {
+      Transforms.delete(editor, {
+        at: matchRange
+      });
+    }
+
     preFormat === null || preFormat === void 0 ? void 0 : preFormat(editor);
 
     if (!format) {
diff --git a/node_modules/@udecode/plate-autoformat/dist/index.js b/node_modules/@udecode/plate-autoformat/dist/index.js
index fb9d8c7..c26b6bb 100644
--- a/node_modules/@udecode/plate-autoformat/dist/index.js
+++ b/node_modules/@udecode/plate-autoformat/dist/index.js
@@ -152,9 +152,13 @@ const autoformatBlock = (editor, {
       if (isBelowSameBlockType) continue;
     }
 
-    slate.Transforms.delete(editor, {
-      at: matchRange
-    });
+    // if the trigger is only 1 char there is nothing to delete, so we'd delete unrelated text
+    if (match.length > 1) {
+      slate.Transforms.delete(editor, {
+        at: matchRange
+      });
+    }
+
     preFormat === null || preFormat === void 0 ? void 0 : preFormat(editor);
 
     if (!format) {
