diff --git a/node_modules/@udecode/plate-autoformat/dist/index.es.js b/node_modules/@udecode/plate-autoformat/dist/index.es.js
index e78e5db..0ba216c 100644
--- a/node_modules/@udecode/plate-autoformat/dist/index.es.js
+++ b/node_modules/@udecode/plate-autoformat/dist/index.es.js
@@ -147,9 +147,14 @@ const autoformatBlock = (editor, {
       if (isBelowSameBlockType) continue;
     }
 
-    deleteText(editor, {
-      at: matchRange
-    });
+
+    // if the trigger is only 1 char there is nothing to delete, so we'd delete unrelated text
+    if (match.length > 1) {
+      deleteText(editor, {
+        at: matchRange
+      });
+    }
+
 
     if (preFormat) {
       preFormat(editor);
diff --git a/node_modules/@udecode/plate-autoformat/dist/index.js b/node_modules/@udecode/plate-autoformat/dist/index.js
index 3debaf1..12b180a 100644
--- a/node_modules/@udecode/plate-autoformat/dist/index.js
+++ b/node_modules/@udecode/plate-autoformat/dist/index.js
@@ -151,9 +151,12 @@ const autoformatBlock = (editor, {
       if (isBelowSameBlockType) continue;
     }
 
-    plateCore.deleteText(editor, {
-      at: matchRange
-    });
+    // if the trigger is only 1 char there is nothing to delete, so we'd delete unrelated text
+    if (match.length > 1) {
+      plateCore.deleteText(editor, {
+        at: matchRange
+      });
+    }
 
     if (preFormat) {
       preFormat(editor);
