diff --git a/node_modules/fraction.js/fraction.js b/node_modules/fraction.js/fraction.js
index 6338725..cd622cc 100644
--- a/node_modules/fraction.js/fraction.js
+++ b/node_modules/fraction.js/fraction.js
@@ -862,7 +862,7 @@
 
       var str = this['s'] === -1 ? "-" : "";
 
-      str += N / D | 0;
+      str += Math.floor(N / D);
 
       N %= D;
       N *= 10;
@@ -873,7 +873,7 @@
       if (cycLen) {
 
         for (var i = cycOff; i--;) {
-          str += N / D | 0;
+          str += Math.floor(N / D);
           N %= D;
           N *= 10;
         }
