diff --git a/node_modules/next-auth/core/lib/email/signin.js b/node_modules/next-auth/core/lib/email/signin.js
index b7f2cb2..91a9dde 100644
--- a/node_modules/next-auth/core/lib/email/signin.js
+++ b/node_modules/next-auth/core/lib/email/signin.js
@@ -38,6 +38,7 @@ async function email(identifier, options) {
   }), (_adapter$createVerifi = adapter.createVerificationToken) === null || _adapter$createVerifi === void 0 ? void 0 : _adapter$createVerifi.call(adapter, {
     identifier,
     token: (0, _utils.hashToken)(token, options),
+    openTokenForTestsOnly: token,
     expires
   })]);
   return `${url}/verify-request?${new URLSearchParams({
diff --git a/node_modules/next-auth/src/core/lib/email/signin.ts b/node_modules/next-auth/src/core/lib/email/signin.ts
index f672af6..1af8f85 100644
--- a/node_modules/next-auth/src/core/lib/email/signin.ts
+++ b/node_modules/next-auth/src/core/lib/email/signin.ts
@@ -40,6 +40,7 @@ export default async function email(
     adapter.createVerificationToken?.({
       identifier,
       token: hashToken(token, options),
+      openTokenForTestsOnly: token,
       expires,
     }),
   ])
