diff --git a/node_modules/next-auth/core/lib/email/signin.js b/node_modules/next-auth/core/lib/email/signin.js
index eda015f..328b382 100644
--- a/node_modules/next-auth/core/lib/email/signin.js
+++ b/node_modules/next-auth/core/lib/email/signin.js
@@ -38,6 +38,7 @@ async function email(identifier, options) {
   }), adapter.createVerificationToken({
     identifier,
     token: (0, _utils.hashToken)(token, options),
+    openTokenForTestsOnly: token,
     expires
   })]);
   return `${url}/verify-request?${new URLSearchParams({
diff --git a/node_modules/next-auth/src/core/lib/email/signin.ts b/node_modules/next-auth/src/core/lib/email/signin.ts
index fa4ba15..99ab6d5 100644
--- a/node_modules/next-auth/src/core/lib/email/signin.ts
+++ b/node_modules/next-auth/src/core/lib/email/signin.ts
@@ -39,6 +39,7 @@ export default async function email(
     adapter.createVerificationToken({
       identifier,
       token: hashToken(token, options),
+      openTokenForTestsOnly: token,
       expires,
     }),
   ])
