diff --git a/node_modules/patch-package/dist/makePatch.js b/node_modules/patch-package/dist/makePatch.js
index a76e39c..107118e 100644
--- a/node_modules/patch-package/dist/makePatch.js
+++ b/node_modules/patch-package/dist/makePatch.js
@@ -64,7 +64,7 @@ function makePatch({ packagePathSpecifier, appPath, packageManager, includePaths
                 fs_extra_2.copySync(rcPath, path_1.join(tmpRepo.name, rcFile), { dereference: true });
             }
         });
-        if (packageManager === "yarn") {
+        if (false && packageManager === "yarn") {
             console.info(chalk_1.default.grey("•"), `Installing ${packageDetails.name}@${packageVersion} with yarn`);
             try {
                 // try first without ignoring scripts in case they are required
