diff --git a/node_modules/react-titled/index.d.ts b/node_modules/react-titled/index.d.ts
new file mode 100644
index 0000000..631e2eb
--- /dev/null
+++ b/node_modules/react-titled/index.d.ts
@@ -0,0 +1,8 @@
+import type { FC, ReactNode } from 'react'
+
+interface TitledProps {
+  readonly title: (parentTitle: string) => string;
+  readonly onChange?: (title: string) => void;
+  readonly children: ReactNode;
+}
+export const Titled: FC<TitledProps>;
