diff --git a/node_modules/slate/dist/index.es.js b/node_modules/slate/dist/index.es.js
index b4b043d..bbdf4fe 100644
--- a/node_modules/slate/dist/index.es.js
+++ b/node_modules/slate/dist/index.es.js
@@ -2429,6 +2429,19 @@ var normalizeNode = (editor, entry) => {
         }
       }
     } else {
+      // If the child is not a text node, and doesn't have a `children` field,
+      // then we have an invalid node that will upset slate.
+      //
+      // eg: `{ type: 'some_node' }`.
+      //
+      // To prevent slate from breaking, we can add the `children` field,
+      // and now that it is valid, we can to many more operations easily,
+      // such as extend normalizers to fix erronous structure.
+      if (!Text.isText(child) && child != null && typeof child === 'object' && !('children' in child)) {
+        const elementChild = child
+        elementChild.children = []
+      }
+
       // Merge adjacent text nodes that are empty or match.
       if (prev != null && Text.isText(prev)) {
         if (Text.equals(_child, prev, {
