diff --git a/node_modules/slate-react/dist/index.es.js b/node_modules/slate-react/dist/index.es.js
index 3d9d640..05b8ad4 100644
--- a/node_modules/slate-react/dist/index.es.js
+++ b/node_modules/slate-react/dist/index.es.js
@@ -2428,7 +2428,15 @@ var ReactEditor = {
       anchor,
       focus
     } = range;
-    return Editor.hasPath(editor, anchor.path) && Editor.hasPath(editor, focus.path);
+    const hasPath = path => {
+      try {
+        Editor.node(editor, path)
+        return true
+      } catch {
+        return false
+      }
+    }
+    return Boolean(hasPath(editor, anchor.path) && hasPath(editor, focus.path));
   }
 
 };
