diff --git a/node_modules/vega-lite/build/src/util.js b/node_modules/vega-lite/build/src/util.js
index 11d689a..07080b5 100644
--- a/node_modules/vega-lite/build/src/util.js
+++ b/node_modules/vega-lite/build/src/util.js
@@ -33,12 +33,6 @@ export function omit(obj, props) {
     }
     return copy;
 }
-/**
- * Monkey patch Set so that `stringify` produces a string representation of sets.
- */
-Set.prototype['toJSON'] = function () {
-    return `Set(${[...this].map(x => stringify(x)).join(',')})`;
-};
 /**
  * Converts any object to a string of limited size, or a number.
  */
diff --git a/node_modules/vega-lite/build/vega-lite.js b/node_modules/vega-lite/build/vega-lite.js
index caffacb..0ab5879 100644
--- a/node_modules/vega-lite/build/vega-lite.js
+++ b/node_modules/vega-lite/build/vega-lite.js
@@ -249,13 +249,6 @@
     return copy;
   }
 
-  /**
-   * Monkey patch Set so that `stringify` produces a string representation of sets.
-   */
-  Set.prototype['toJSON'] = function () {
-    return `Set(${[...this].map(x => stringify(x)).join(',')})`;
-  };
-
   /**
    * Converts any object to a string of limited size, or a number.
    */
